Sources/ApodiniNetworking/ChannelHandlers/UpgradeHandler.swift

```diff
@@ -0,0 +1,149 @@
+//
+// This source file is part of the Apodini open source project
+//
+// SPDX-FileCopyrightText: 2019-2021 Paul Schmiedmayer and the Apodini project authors (see CONTRIBUTORS.md) <paul.schmiedmayer@tum.de>
+//
+// SPDX-License-Identifier: MIT
+//
+
+import NIO
+import NIOHTTP1
+import NIOHTTP2
+import NIOWebSocket
+import WebSocketKit
+import Logging
+
+
+/// A removable channel handler which can be used to upgrade HTTP1 requests to a different protocol.
+/// Heavily insired by Vapor's equivalent of this
```

Comment:
Might want to check the Vapor Licensing if there are any restrictions or duties, like including the original license info.

Reply:
@PSchmiedmayer what's your take on this, wrt to the Licensing?

Generally speaking, I didn't copy any code from Vapor, but at the end of the day there's only so many ways you can implement a HTTP server so there's bound to be some overlap.
In this specific case I did in fact (to a certain extent, mainly in terms of handling the protocol upgrade w/in the NIO handler) base my implementation on Vapor's. How should this be indicated?

Reply:
@lukaskollmer @Supereg Yes, I would suggest to add a reference to the Vapor License similar to how we did it in the Application+... extensions. I don't think there is any harm in giving credit to other open-source work and even if you did not direly copy code I would somehow consider this an "indirect citation" and think we can credit the Vapor authors for doing thorough the struggles of implementing the functionality first and giving us a change to use their implementation as a reference.